tests/test_io.py

```diff
@@ -47,6 +47,28 @@ def test_io_xlsx(test_df, meta_args):
         os.remove(file)
 
 
+def test_init_df_with_na_unit(test_pd_df):
```

Comment:
you can use the `tmpdir` fixture (https://docs.pytest.org/en/stable/tmpdir.html) to avoid the os.remove calls here

Reply:
Thanks @znicholls, nice suggestion! When trying to implement this, I realized that the IamDataFrame can't be initialized with a file as `Path` or `LocalPath`, so I started working on that, and cleaned up the sequence of calls in the initalization... and quite a rabbit hole later, I realized that it makes more sense to put that in a new PR.